lib/jsonApi.js

```diff
@@ -16,12 +16,22 @@ jsonApi.mockHandlers = mockHandlers.handlers;
 jsonApi.setConfig = function(apiConfig) {
   jsonApi._apiConfig = apiConfig;
   jsonApi._apiConfig.base = jsonApi._cleanBaseUrl(jsonApi._apiConfig.base);
-  responseHelper.setBaseUrl(jsonApi._apiConfig.base);
+  jsonApi._apiConfig.pathPrefix = jsonApi._concatenateUrlPrefix(jsonApi._apiConfig);
+  responseHelper.setBaseUrl(jsonApi._apiConfig.pathPrefix);
   responseHelper.setMetadata(jsonApi._apiConfig.meta);
 };
 
 jsonApi.authenticate = router.authenticateWith;
 
+jsonApi._concatenateUrlPrefix = function(config) {
+  var pathPrefix = "";
+  pathPrefix += (config.protocol || "https") + "://";
+  pathPrefix += (config.hostname || "localhost");
+  if (config.port) pathPrefix += ":" + config.port;
+  pathPrefix += config.base;
+  return pathPrefix;
```

Comment:
Would it be worth using the standard library [`url`](https://nodejs.org/api/url.html) module here to build a url object and then format it rather than concatenating it manually?